fab/sampling_methods/transition_operators/metropolis.py

```diff
@@ -47,6 +47,8 @@ def transition(self, x: torch.Tensor, log_p_x: LogProbFunc, i: int) -> torch.Ten
             acceptance_probability = torch.exp(x_proposed_log_prob - x_prev_log_prob)
             # not that sometimes this will be greater than one, corresonding to 100% probability of
             # acceptance
+            acceptance_probability = torch.nan_to_num(acceptance_probability, nan=0.0, posinf=0.0,
```

Comment:
Prevent Nans/infs from effecting the acceptance probability in the wrong way (typically these result from too big a step size, so we treat them in a way that generally decreases the step size).